rest/admin_api_test.go

```diff
@@ -800,60 +800,45 @@ func TestSessionTtlGreaterThan30Days(t *testing.T) {
-	assert.NoError(t, err)
-
-	user, err = a.GetUser("")
-	assert.NoError(t, err)
-	goassert.True(t, user.Disabled())
-
-	response := rt.SendRequest("PUT", "/db/doc", `{"hi": "there"}`)
-	assertStatus(t, response, 401)
-
-	user, err = a.NewUser("pupshaw", "letmein", channels.SetOf(t, "*"))
-	a.Save(user)
-
-	assertStatus(t, rt.SendAdminRequest("GET", "/db/_session", ""), 200)
-
-	response = rt.SendAdminRequest("POST", "/db/_session", `{"name":"pupshaw", "ttl":1}`)
-	assertStatus(t, response, 200)
-
-	var body db.Body
-	base.JSONUnmarshal(response.Body.Bytes(), &body)
-	sessionId := body["session_id"].(string)
-	sessionExpiration := body["expires"].(string)
-	assert.NotEmpty(t, sessionId)
-	assert.NotEmpty(t, sessionExpiration)
-	assert.True(t, body["cookie_name"].(string) == "SyncGatewaySession")
+	// Fake session with more than 10% of the 24 hours TTL has elapsed. It should cause a new
+	// cookie to be sent by the server with the same session ID and an extended expiration date.
+	fakeSession := auth.LoginSession{
+		ID:         base.GenerateRandomSecret(),
+		Username:   "Alice",
+		Expiration: time.Now().Add(4 * time.Hour),
+		Ttl:        24 * time.Hour,
+	}
 
+	assert.NoError(t, rt.Bucket().Set(auth.DocIDForSession(fakeSession.ID), 0, fakeSession))
 	reqHeaders := map[string]string{
-		"Cookie": "SyncGatewaySession=" + body["session_id"].(string),
+		"Cookie": auth.DefaultCookieName + "=" + fakeSession.ID,
 	}
-	response = rt.SendRequestWithHeaders("PUT", "/db/doc1", `{"hi": "there"}`, reqHeaders)
-	assertStatus(t, response, 201)
 
-	goassert.True(t, response.Header().Get("Set-Cookie") == "")
+	response := rt.SendRequestWithHeaders("PUT", "/db/doc1", `{"hi": "there"}`, reqHeaders)
+	log.Printf("PUT Request: Set-Cookie: %v", response.Header().Get("Set-Cookie"))
+	assertStatus(t, response, http.StatusCreated)
+	assert.Contains(t, response.Header().Get("Set-Cookie"), auth.DefaultCookieName+"="+fakeSession.ID)
+
+	response = rt.SendRequestWithHeaders("GET", "/db/doc1", "", reqHeaders)
+	log.Printf("GET Request: Set-Cookie: %v", response.Header().Get("Set-Cookie"))
+	assertStatus(t, response, http.StatusOK)
+	assert.Equal(t, "", response.Header().Get("Set-Cookie"))
+
+	fakeSession.Expiration = time.Now().Add(-2 * time.Hour)
+	assert.NoError(t, rt.Bucket().Set(auth.DocIDForSession(fakeSession.ID), 0, fakeSession))
+	log.Printf("Expired fake session: %v", fakeSession)
 
-	//Sleep for 150ms seconds, this will ensure 10% of the 1 second session ttl has elapsed and
-	//should cause a new Cookie to be sent by the server with the same session ID and an extended expiration date
-	time.Sleep(150 * time.Millisecond)
-	response = rt.SendRequestWithHeaders("PUT", "/db/doc2", `{"hi": "there"}`, reqHeaders)
-	assertStatus(t, response, 201)
+	response = rt.SendRequestWithHeaders("GET", "/db/doc1", "", reqHeaders)
+	log.Printf("GET Request: Set-Cookie: %v", response.Header().Get("Set-Cookie"))
+	assert.NotEqual(t, "", response.Header().Get("Set-Cookie"))
```

Comment:
As discussed in other comment, we're relying on CB Server to delete the document via TTL to get expired sessions.

As you've found out, if we somehow end up with an expired session in the bucket that is not removed, it actually extends the session! This doesn't seem like correct behaviour. We can sync up on this today to figure out next steps.

Reply:
Sure, thank you Ben